pkgs/stdenv/darwin/default.nix

```diff
@@ -59,11 +59,12 @@ in rec {
   stageFun = step: last: {shell             ? "${bootstrapTools}/bin/bash",
                           overrides         ? (self: super: {}),
                           extraPreHook      ? "",
+                          extraNativeBuildInputs,
                           extraBuildInputs,
```

Comment:
Also, instead of just setting `allowedRequisites` in the final stdenv, we'll have the opportunity to set it in intermediate derivations based just on  `buildInputs`/`propagatedBuildInputs`. This will make bootstrapping more maintainable, cc @copumpkin.